lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs

```diff
@@ -191,7 +191,12 @@ public IEnumerable<Uri> GetChildren(Uri directoryUri)
         /// <returns>The URI to the remote file</returns>
         public Uri CreateUriForPath(string path)
         {
-            return new Uri(_client.BaseUri.AbsoluteUri.TrimEnd('/') + '/' + path.Trim('/'));
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path), "Specified path is null");
+            }
+            var uriPrefix = GetUriPrefix();
+            return path.StartsWith(uriPrefix) ? new Uri(path) : new Uri(uriPrefix + '/' + path.Trim('/'));
```

Comment:
Can this be new Uri(uriPrefix, path)? Also you can try to create a Uri from the path. If the path is relative it will throw a UriFormatException. If that is thrown, you can catch and retry with Uri(uriPrefix, path).

Reply:
Scott has suggsted to do something like this
try
{
   testUri = new Uri(path);
}
catch(System.UriFormatExecption)
{
   testUri = new Uri(GetBaseUri(), path);
}

Reply:
Followed this suggestion and updated pull request.